wlog-zerolog/internal/logger.go

```diff
@@ -158,54 +158,35 @@ func (e *zeroLogEntry) Evt() *zerolog.Event {
 
 type zeroLogger struct {
 	logger zerolog.Logger
-	level  zerolog.Level
-}
-
-func (l *zeroLogger) should(level zerolog.Level) bool {
-	if level < l.level {
-		return false
-	}
-	return true
+	wlog.AtomicLogLevel
 }
 
 func (l *zeroLogger) Log(params ...wlog.Param) {
-	if !l.should(zerolog.NoLevel) {
-		return
-	}
 	logOutput(l.logger.Log, "", params)
 }
 
 func (l *zeroLogger) Debug(msg string, params ...wlog.Param) {
-	if !l.should(zerolog.DebugLevel) {
-		return
+	if l.Enabled(wlog.DebugLevel) {
+		logOutput(l.logger.Log, msg, params)
 	}
-	logOutput(l.logger.Log, msg, params)
 }
 
 func (l *zeroLogger) Info(msg string, params ...wlog.Param) {
-	if !l.should(zerolog.InfoLevel) {
-		return
+	if l.Enabled(wlog.InfoLevel) {
+		logOutput(l.logger.Log, msg, params)
 	}
-	logOutput(l.logger.Log, msg, params)
 }
 
 func (l *zeroLogger) Warn(msg string, params ...wlog.Param) {
-	if !l.should(zerolog.WarnLevel) {
-		return
+	if l.Enabled(wlog.WarnLevel) {
+		logOutput(l.logger.Log, msg, params)
 	}
-	logOutput(l.logger.Log, msg, params)
 }
 
 func (l *zeroLogger) Error(msg string, params ...wlog.Param) {
-	if !l.should(zerolog.ErrorLevel) {
-		return
+	if l.Enabled(wlog.ErrorLevel) {
+		logOutput(l.logger.Log, msg, params)
 	}
-	logOutput(l.logger.Log, msg, params)
-}
-
-func (l *zeroLogger) SetLevel(level wlog.LogLevel) {
```

Comment:
Rubber-ducking here, but is it the case that it's safe to remove this exported function because the function is actually defined on an unexported struct (`*zerologger`) and is not part of any exported interface implemented by the struct? Just want to verify/understand whether this is an API break that we just think is minor or if it's not.

Reply:
Yeah, this is not a break because the embedded AtomicLogLevel provides this method (as well as `Enabled()`). If that were not true, `*zeroLogger` would not implement `wlog.LeveledLogger` which requires `SetLevel`